entity-service-impl/src/main/java/org/hypertrace/entity/metric/EntityCounterMetricProvider.java

```diff
@@ -0,0 +1,102 @@
+package org.hypertrace.entity.metric;
+
+import io.micrometer.core.instrument.Counter;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.AllArgsConstructor;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+import org.hypertrace.entity.service.change.event.impl.ChangeResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EntityCounterMetricProvider {
```

Comment:
naming nit - maybe metric sender? A provider gives you something, this class doesn't.